File: Client/src/models/sourceDocument.ts
import type { ContentAccessLevel, ReviewStatus } from './enums'

export interface SourceDocument {
  id: string
  courseId: string
  title: string
  fileName: string
  filePath: string
  sourceName: string
  sourcePublishedAt?: string | null
  sourceUpdatedAt?: string | null
  pageCount: number
  textExtractedAt?: string | null
  reviewStatus: ReviewStatus
  accessLevel: ContentAccessLevel
  reviewedBy?: string | null
  reviewedAt?: string | null
  reviewComment?: string | null
}

export interface SourceDocumentText {
  id: string
  courseId: string
  title: string
  pageCount: number
  textExtractedAt?: string | null
  extractedText: string
}

File: spk/API/Entities/SourceDocument.cs
namespace API.Entities;

public class SourceDocument : ModeratedEntity
{
    public Guid CourseId { get; set; }

    public Course? Course { get; set; }

    public string Title { get; set; } = string.Empty;

    public string FileName { get; set; } = string.Empty;

    public string FilePath { get; set; } = string.Empty;

    public string SourceName { get; set; } = string.Empty;

    public DateOnly? SourcePublishedAt { get; set; }

    public DateOnly? SourceUpdatedAt { get; set; }

    public int PageCount { get; set; }

    public string? ExtractedText { get; set; }

    public DateTime? TextExtractedAt { get; set; }
}
